src/SqlAsyncCollector.cs

```diff
@@ -318,6 +360,10 @@ public class TableInformation
             /// </summary>
             public string Query { get; }
 
+            /// <summary>
+            /// All of the column names, for SQL to use when data is passed in as JObject
+            /// </summary>
+            public static IEnumerable<string> ColumnNames { get; set; }
```

Comment:
This shouldn't be static - we can have multiple executions of this which may have different columns running at the same time.

And really this should probably be a member on TableInfo or something - the async collector doesn't seem like the correct place to be storing information about the table being upserted to.